dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/view/ViewUtil.java

```diff
@@ -0,0 +1,167 @@
+            return label;
+        }
+
+        public String getHint() {
+            return hint;
+        }
+
+        @Override
+        public String toString() {
+            return "SecretHolder{" +
+                    "name='" + name + '\'' +
+                    ", value='" + value + '\'' +
+                    ", label='" + label + '\'' +
+                    ", hint='" + hint + '\'' +
+                    '}';
+        }
+    }
+
+
+    static String interpolateValues(final String inputJson) {
+        final StackContext context = ViewUtil.currentStackContext();
+        if (null == context) {
+            throw new IllegalStateException(
+                    "This method can only be called once a context has been previously created.");
+        }
+        try {
+            final RuntimeServices runtimeServices = RuntimeSingleton.getRuntimeServices();
+            final StringReader stringReader = new StringReader(inputJson);
+            final SimpleNode simpleNode = runtimeServices.parse(stringReader, "app template");
+
+            final Template template = new Template();
+            template.setData(simpleNode);
+            template.initDocument();
+
+            final VelocityContext velocityContext = new VelocityContext();
+
+            velocityContext.put("app", context.appView);
+
+            final List<SiteView> sites = context.appView.getSites();
+            if (null != sites) {
+                final Map<String, SiteView> sitesById = sites.stream()
+                        .collect(Collectors.toMap(SiteView::getId, Function.identity()));
+
+                if (null != context.currentSite) {
+
+                    final SiteView siteView = sitesById.get(context.currentSite);
+
+                    velocityContext.put("siteId", context.currentSite);
+                    velocityContext.put("siteName", siteView.getName());
+                    velocityContext.put("configured", siteView.isConfigured());
+
+                    final List<Map<String, Object>> currentSiteSecrets = context.secretsBySite
+                            .get(context.currentSite);
+                    if (null != currentSiteSecrets) {
+                        for (final Map<String, Object> siteSecret : currentSiteSecrets) {
+                            final String name = (String) siteSecret.get("name");
+                            final String hint = (String) siteSecret.get("hint");
+                            final String label = (String) siteSecret.get("label");
+                            final Object value = siteSecret.get("value");
+                            velocityContext.put(name, new SecretHolder(name, label, hint, value));
+                        }
+                    }
+                }
+            }
+
+            final StringWriter stringWriter = new StringWriter();
+            template.merge(velocityContext, stringWriter);
+            return stringWriter.toString();
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392890)